src/lib.rs

```diff
@@ -28,23 +28,57 @@ use errors::Error;
 use hashlock::{sha256, Preimage};
 
 use bitcoin::{
-    sighash::Prevouts, transaction::InputWeightPrediction, OutPoint, Transaction, TxIn, TxOut,
+    secp256k1::XOnlyPublicKey as BitcoinXOnly, sighash::Prevouts,
+    transaction::InputWeightPrediction, OutPoint, Transaction, TxIn, TxOut,
+};
+use musig2::{
+    secp256k1::XOnlyPublicKey as Musig2XOnly, AdaptorSignature, AggNonce, CompactSignature,
+    PartialSignature, PubNonce, SecNonce,
 };
-use musig2::{AdaptorSignature, AggNonce, CompactSignature, PartialSignature, PubNonce, SecNonce};
 use secp::{MaybeScalar, Point, Scalar};
 use serde::{Deserialize, Serialize};
 
-use std::{
-    borrow::Borrow,
-    collections::{BTreeMap, BTreeSet},
-};
-
 pub use contract::{
     ContractParameters, Outcome, OutcomeIndex, PayoutWeights, PlayerIndex, SigMap, WinCondition,
 };
 pub use oracles::{attestation_locking_point, attestation_secret, EventLockingConditions};
 pub use parties::{MarketMaker, Player};
 
+use std::{
+    borrow::Borrow,
+    collections::{BTreeMap, BTreeSet},
+};
+
+/// Used to convert musig2 xonly key into bitcoin xonly key only needed
+/// until crate bitcoin updates secp256k1 to v0.30
+/// will be removed/deprecated once the dependency has upgrade
+pub fn convert_xonly_key(key: Musig2XOnly) -> BitcoinXOnly {
+    let serialized = key.serialize();
+
+    // Create a fixed size array for the x-coordinate
+    let x_bytes: [u8; 32] = if serialized.len() == 32 {
+        serialized.try_into().expect("Length already checked")
+    } else if serialized.len() == 33 {
+        serialized[1..]
+            .try_into()
+            .expect("Slice to array conversion failed")
+    } else {
+        panic!("Unexpected key length: {}", serialized.len())
+    };
+
+    BitcoinXOnly::from_slice(&x_bytes).expect("Valid key")
+}
+
+/// Used to convert secp point into bitcoin xonly key
+/// only needed until crate bitcoin updates secp256k1 to v0.30
+/// will be removed/deprecated once the dependency has upgrade
+pub fn convert_point(key: Point) -> BitcoinXOnly {
+    let serialized = key.serialize(); // This gives us 33 bytes (compressed format)
+                                      // Skip the first byte (compression prefix) to get just the x-coordinate
+    let x_only_bytes = &serialized[1..];
+    BitcoinXOnly::from_slice(x_only_bytes).expect("Valid key")
```

Comment:
A simpler way is available thanks to [`secp::Point::serialize_xonly`](https://docs.rs/secp/0.4.2/secp/struct.Point.html#method.serialize_xonly)

```suggestion
    BitcoinXOnly::from_slice(&key.serialize_xonly()).expect("Valid key")
```

Reply:
so oddly enough I think this is again affected by the "serdect" crate change, if we try to using 
`BitcoinXOnly::from_slice(&key.serialize_xonly()).expect("Valid key")` the test always report an invalid pubkey:
```
hread 'regtest::stress_test' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- regtest::all_players_cooperate stdout ----
thread 'regtest::all_players_cooperate' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey

---- regtest::contract_expiry_all_winners_cooperate stdout ----
thread 'regtest::contract_expiry_all_winners_cooperate' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey

---- regtest::with_on_chain_resolutions stdout ----
thread 'regtest::with_on_chain_resolutions' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey

---- regtest::market_maker_reclaims_outcome_tx stdout ----
thread 'regtest::market_maker_reclaims_outcome_tx' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey

---- regtest::all_winners_cooperate stdout ----
thread 'regtest::all_winners_cooperate' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey

---- regtest::individual_sellback stdout ----
thread 'regtest::individual_sellback' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey

---- regtest::contract_expiry_on_chain_resolution stdout ----
thread 'regtest::contract_expiry_on_chain_resolution' panicked at src/lib.rs:66:48:
Valid key: InvalidPublicKey
```
Let me know what the preferred solution is here

Reply:
I think you may be conflating two unrelated errors. `InvalidPublicKey` is a specific error value, whereas if this code was failing, it'd panic

Reply:
That's bizarre, it works fine for me. here's my diff on your latest commit:

```diff
diff --git a/src/lib.rs b/src/lib.rs
index 537c72f..616d266 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -60,9 +60,7 @@ pub fn convert_xonly_key(key: Musig2XOnly) -> BitcoinXOnly {
 /// only needed until crate bitcoin updates secp256k1 to v0.30
 /// will be removed/deprecated once the dependency has upgraded
 pub fn convert_point(key: Point) -> BitcoinXOnly {
-    let serialized = key.serialize();
-    let x_only_bytes = &serialized[1..];
-    BitcoinXOnly::from_slice(&x_only_bytes).expect("Valid key")
+    BitcoinXOnly::from_slice(&key.serialize_xonly()).expect("Valid key")
 }
 
 /// Represents the combined output of building all transactions and precomputing
diff --git a/src/regtest.rs b/src/regtest.rs
index c1b126b..e16a9be 100644
--- a/src/regtest.rs
+++ b/src/regtest.rs
@@ -84,6 +84,7 @@ fn run_bitcoind() -> Option<(BitcoindSubprocessHandle, BitcoinClient)> {
     let child: process::Child = process::Command::new("bitcoind")
         .arg("-regtest")
         .arg("-server")
+        .arg("-listen=0")
         .arg(format!("-rpcport={}", rpc_port))
         .arg(format!("-port={}", p2p_port))
         .arg(format!("-rpcuser={}", DEFAULT_REGTEST_RPC_USERNAME))
```

Are you sure you're using `key.serialize_xonly()` and not `key.serialize()`? The former returns a `[u8; 32]`, the latter returns `[u8; 33]`.